sopel/modules/choose.py

```diff
@@ -12,8 +12,51 @@
 from __future__ import absolute_import, division, print_function, unicode_literals
 
 import random
+import unicodedata
 
-from sopel import plugin
+from sopel import formatting, plugin
+
+
+def _format_safe(text):
+    """Remove excess whitespace and terminate IRC formatting.
+
+    :param str text: text to clean of whitespace
+    :rtype: str
+
+    Our own take on ``str.strip()`` that skips stripping off IRC formatting
+    and makes sure any formatting codes are closed if necessary.
+    """
+    start = end = 0
```

Comment:
```suggestion
    if not text:
        return ''

    start = end = 0
```

This is a minor optimization, but heh, worth a shot.

Reply:
I like this idea, but maybe not the exact implementation. If `text` is a falsey non-string value, it should fail—e.g. this would return `''` for `_format_safe(None)`, a case that should raise `TypeError` or `ValueError`. Since I already have a couple more test tweaks in mind, I can do something about this at the same time.

Reply:
I'm, like: yeah, OK. But really I'm like: yeah, whatever.

Also, the very sassy me would say: this wouldn't happen with a proper static type system. Which is true, but not helpful at all.

Reply:
I think you'll like the commit message on 0c763cf then. 😏